src/tools/kubectl-delete.ts

```diff
@@ -51,11 +48,10 @@ export const kubectlDeleteSchema = {
       },
       gracePeriodSeconds: {
         type: "number",
-        description: "Period of time in seconds given to the resource to terminate gracefully",
-        optional: true
+        description: "Period of time in seconds given to the resource to terminate gracefully"
       }
     },
-    required: [],
+    required: ["labelSelector", "manifest", "filename", "gracePeriodSeconds"],
```

Comment:
```suggestion
    required: ["resourceType", "name", "namespace"],
```

If you're removing optional then the ones that aren't optional should be in required. This is an error in each of the ones below too.